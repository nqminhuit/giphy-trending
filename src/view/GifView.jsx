import PropTypes from "prop-types";
import React from "react";
import { useHistory } from "react-router";

// TODO:Introducing Error Boundaries: https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html
export default function GifView() {
  const history = useHistory();
  if (!history.location) {
    throw new Error("location not found");
  }

  const { state } = history.location;
  if (!state) {
    throw new Error("state not found");
  }

  const { gifOriginUrl } = state;
  return (
    <div className="w-100 h-100 bg-dark position-fixed d-flex">
      <img className="img-fluid m-auto" src={gifOriginUrl} alt="giphy gif" />
    </div>
  );

}

GifView.propTypes = {
  gifUrl: PropTypes.string,
};
