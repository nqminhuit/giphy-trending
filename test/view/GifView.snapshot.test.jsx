import React from "react";
import renderer from "react-test-renderer";
import GifView from "../../src/view/GifView.jsx";

jest.mock("react-router", () => ({
  useHistory: () => ({ location: { state: { gifOriginUrl: "https://www.giphy.com" } } })
}));

it("should render <GifView />", () => {
  const component = renderer.create(<GifView />).toJSON();
  expect(component).toMatchSnapshot();
});
