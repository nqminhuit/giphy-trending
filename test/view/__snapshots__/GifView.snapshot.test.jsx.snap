// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <GifView /> 1`] = `
<div
  className="w-100 h-100 bg-dark position-fixed d-flex"
>
  <img
    alt="giphy gif"
    className="img-fluid m-auto"
    src="https://www.giphy.com"
  />
</div>
`;
